src/ucp/core/ucp_mm.c

```diff
@@ -850,52 +921,61 @@ static ucs_status_t ucp_advice2uct(unsigned ucp_advice, uct_mem_advice_t *uct_ad
     return UCS_ERR_INVALID_PARAM;
 }
 
+static ucs_status_t ucp_memh_uct_mem_advise(ucp_context_h context,
+                                            ucp_mem_h memh,
+                                            ucp_md_index_t md_index,
+                                            ucp_md_index_t uct_memh_offset,
```

Comment:
Why do you need offset here? do you expect to be called for exported keys?

Reply:
also would not be having a separate array for exported keys be more clear than having 2*num_mds array in memh?

Reply:
> removed, needed in the PR which adds EXPORT_MEMH

Reply:
> also would not be having a separate array for exported keys be more clear than having 2*num_mds array in memh?

it is necessary to discuss it within the team